packages/smart-accounts/src/mav-2/accounts/base.ts

```diff
@@ -0,0 +1,480 @@
+        typeof td.domain === "object" &&
+        "verifyingContract" in td.domain &&
+        td.domain.verifyingContract === accountAddress;
+
+      if (owner.type === "webAuthn") {
+        const hash = hashTypedData(
+          toReplaySafeTypedData({
+            chainId: client.chain.id,
+            address: getDefaultWebauthnValidationModuleAddress(client.chain),
+            hash: hashTypedData(td),
+            salt: concatHex([`0x${"00".repeat(12)}`, accountAddress]),
+          }),
+        );
+        const validationSignature = toWebAuthnSignature(
+          await owner.sign({ hash }),
+        );
+        return isDeferredAction
+          ? pack1271Signature({
+              validationSignature,
+              entityId,
+            })
+          : validationSignature;
+      }
+
+      const { data } = await prepareSignature({
+        type: "eth_signTypedData_v4",
+        data: td as TypedDataDefinition, // TODO(jh): why do we need to cast?
+      });
+
+      const signature = await signTypedData(client, {
+        ...data,
+        account: owner,
+      });
+
+      return isDeferredAction
+        ? concat([SignaturePrefix.EOA, signature])
+        : formatSignature(signature);
+    },
+
+    async signUserOperation(uo) {
+      const hash = getUserOperationHash({
+        chainId: uo.chainId ?? client.chain.id,
+        entryPointAddress: entryPoint.address,
+        entryPointVersion: entryPoint.version,
+        userOperation: {
+          ...uo,
+          sender: accountAddress,
+        },
+      });
+
+      if (owner.type === "webAuthn") {
+        const validationSignature = toWebAuthnSignature(
+          await owner.sign({ hash }),
+        );
+        const signature = deferredActionData
+          ? concatHex([deferredActionData, validationSignature])
+          : validationSignature;
+        deferredActionData = undefined;
+        return concatHex(["0xff", signature]);
+      }
+
+      const validationSignature = await signMessage(client, {
+        account: owner,
+        message: { raw: hash },
+      });
+
+      const packedSignature = packUOSignature({
+        validationSignature,
+      });
+
+      const signature = deferredActionData
+        ? concatHex([deferredActionData, packedSignature])
+        : packedSignature;
+
+      deferredActionData = undefined;
+
+      return signature;
+    },
+
+    // TODO(jh): should we handle different gas estimation stuff for 7702 or webauthn here?
+    // or will the bundler client handle that later?
+    // https://github.com/wevm/viem/blob/main/src/account-abstraction/accounts/implementations/toCoinbaseSmartAccount.ts
+    // userOperation: {
+    //   estimateGas: () => {
+    //     TODO
+    //   }
+    // }
+
+    // TODO(jh): figure out how to properly use the generic param for 7702
+    // https://github.com/wevm/viem/blob/f82cdd929062c0d079142de1555a472f78ac59e1/src/account-abstraction/accounts/types.ts#L204
+
+    extend: {
```

Comment:
`prepareSignature` and `formatSignature` do need to be here for wallet-server to consumes